aspnet/mvc/overview.rst

```diff
@@ -1,8 +1,313 @@
+
+ASP.NET Core MVC gives you a powerful, patterns-based way to build dynamic websites that enables a clean separation of concerns and gives you full control over markup for enjoyable, agile development. ASP.NET Core MVC includes many features that enable fast, TDD-friendly development for creating sophisticated applications that use the latest web standards.
+
+ASP.NET Core MVC in ASP.NET Core includes support for building web pages and HTTP services in a single aligned framework that can be hosted in IIS or self-hosted in your own process.
+
+Features
+--------
+
+ASP.NET Core MVC includes the following features:
+  - `Routing`_
+  - `Model binding`_
+  - `Model validation`_
+  - `Dependency injection`_
+  - `Filters`_
+  - `Areas`_
+  - `Web APIs`_
+  - `Testability`_
+  - `Razor view engine`_
+  - `Strongly typed views`_
+  - `Mobile support`_
+
+Routing
+^^^^^^^
+
+ASP.NET Core MVC is built on top of :doc:`ASP.NET Core's routing </fundamentals/routing>`, a powerful URL-mapping component that lets you build applications that have comprehensible and searchable URLs. This enables you to define your application's URL naming patterns that work well for search engine optimization (SEO) and for link generation, without regard for how the files on your web server are organized. You can define your routes using a convenient route template syntax that supports route value constraints, defaults and optional values.
+
+*Convention-based routing* enables you to globally define the URL formats that your application accepts and how each of those formats maps to a specific action method on given controller. When an incoming request is received, the routing engine parses the incoming URL and matches it to one of the defined URL formats, and calls the associated controller's action method.
+
+
+.. code-block:: c#
+
+  routes.MapRoute(name: "Default", template: "{controller=Home}/{action=Index}/{id?}");
+
+*Attribute routing* enables you to specify routing information by decorating your controllers and actions with attributes that define your application's routes. This means that your route definitions are placed next to the controller and action with which they're associated.
+
+.. code-block:: c#
+  :emphasize-lines: 1,4
+
+  [Route("api/[controller]")]
+  public class ProductsController : Controller
+  {
+    [HttpGet("{id}")]
+    public IActionResult GetProduct(int id)
+    {
+      ...
+    }
+  }
+
+Model binding
+^^^^^^^^^^^^^
+
+ASP.NET Core MVC :doc:`model binding </mvc/models/model-binding>` converts client request data  (form values, route data, query string parameters, HTTP headers) into objects that the controller can handle. As a result, your controller logic doesn't have to do the work of figuring out the incoming request data; it simply has the data as parameters to its action methods.
```

Comment:
A model binder in MVC provides a simple way to map posted form values to a .NET Core type and pass the type to an action method as a parameter. Binders also give you control over the deserialization of types that are passed to action methods. Model binders are like type converters, because they can convert HTTP requests into objects that are passed to an action method. However, they also have information about the current controller context.